pyproject.toml

```diff
@@ -1,2 +1,58 @@
 [tool.black]
 target-version = ['py310']
+
+[tool.poetry]
+name = "screenpy"
+version = "4.0.2"
+description = "Screenplay pattern base for Python automated UI test suites."
+authors = ["Perry Goy <perry.goy@gmail.com>"]
+license = "MIT"
+#packages = []
+repository = "https://github.com/ScreenPyHQ/screenpy"
+documentation = "https://screenpy-docs.readthedocs.io"
+readme = "README.md"
+classifiers = [
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Quality Assurance",
+    "Topic :: Software Development :: Testing",
+    "Topic :: Software Development :: Testing :: BDD",
+    "Natural Language :: English",
+    "License :: OSI Approved :: MIT License",
+]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+PyHamcrest = "~2.0"  # ">=2.0.0,<2.1"
+typing-extensions = ">=4.1.1,<4.4"
+
+[tool.poetry.dev-dependencies]
+pre-commit = "^2.17.0"
+pytest = "^7.1.1"
+pylint = "^2.12.2"
+mypy = "^0.941"
+black = "^22.1.0"
+isort = "^5.10.1"
+flake8 = "^4.0.1"
+coverage = "*"
```

Comment:
I noticed `tox.ini` uses `coverage`, so I figured it probably needs to be included here, but not sure if we need to limit it to a specific version.

Reply:
`tox` does its own dependency installation, but being able to actually _read_ the coverage report would need `coverage`, so... yeah let's add it. Haha.

Reply:
I need to look for a way to combine the versions in pyproject and precommit

Reply:
After a long conversation with @perrygoy, he convinced me that any local installation of linters does not matter. What matters is the versions in the pre-commit (and eventually in the CI/CD checking).  
So we don't need to worry about syncing the versions of optional packages installed in a dev environment.  I've made them completely optional and purely for convenience.